src/irr/asset/CGLTFLoader.cpp

```diff
@@ -50,6 +50,203 @@ namespace irr
 				cpuBuffers.emplace_back() = core::smart_refctd_ptr<ICPUBuffer>(cpuBuffer);
 			}
 
+			core::vector<core::smart_refctd_ptr<CCPUMesh>> cpuMeshes;
+			for (auto& glTFnode : glTF.nodes) 
+			{
+				auto cpuMesh = cpuMeshes.emplace_back();
+
+				for (auto& glTFprimitive : glTFnode.glTFMesh.primitives) 
+				{
+					typedef std::remove_reference<decltype(glTFprimitive)>::type SGLTFPrimitive;
+					
+					auto meshBuffer = core::make_smart_refctd_ptr<asset::ICPUMeshBuffer>();
+
+					auto getMode = [&](uint32_t modeValue) -> E_PRIMITIVE_TOPOLOGY
+					{
+						switch (modeValue)
+						{
+							case SGLTFPrimitive::SGLTFPT_POINTS:
+								return EPT_POINT_LIST;
+							case SGLTFPrimitive::SGLTFPT_LINES:
+								return EPT_LINE_LIST;
+							case SGLTFPrimitive::SGLTFPT_LINE_LOOP:
+								return EPT_LINE_LIST_WITH_ADJACENCY; // check it
+							case SGLTFPrimitive::SGLTFPT_LINE_STRIP:
+								return EPT_LINE_STRIP;
+							case SGLTFPrimitive::SGLTFPT_TRIANGLES:
+								return EPT_TRIANGLE_LIST;
+							case SGLTFPrimitive::SGLTFPT_TRIANGLE_STRIP:
+								return EPT_TRIANGLE_STRIP;
+							case SGLTFPrimitive::SGLTFPT_TRIANGLE_FAN:
+								return EPT_TRIANGLE_STRIP_WITH_ADJACENCY; // check it
+						}
+					};
+
+					const E_PRIMITIVE_TOPOLOGY primitiveTopology = getMode(glTFprimitive.mode.value());
+
+					if (glTFprimitive.indices.has_value())
+					{	
+						auto& glTFIndexAccessor = glTFprimitive.accessors["INDEX"];
+
+						// TODO
+					}
+
+					auto statusPosition = glTFprimitive.accessors.find("POSITION");
+					if (statusPosition != glTFprimitive.accessors.end())
+					{
+						// TODO: make it a common lambda for handling accessors
+
+						auto& glTFPositionAccessor = glTFprimitive.accessors["POSITION"];
+
+						typedef std::remove_reference<decltype(glTFPositionAccessor)>::type SGLTFAccessor;
+
+						auto getType = [&](std::string type)
```

Comment:
you will need to join the two lambdas into one and take a pair of `type` and `componentType`

because our `E_FORMAT` combines both concepts into a single enum

Reply:
what about matrices?

Reply:
https://github.com/KhronosGroup/glTF/tree/master/specification/2.0#accessor-element-size

Reply:
We dont support that as a vertex attribute, anything like that needs to be fed in as 4x consecutive attribute ids of and RGBA format

I think a matrix accessor is mainly for the bones/nodes/animation keyframes.

Reply:
you could do it as `E_FORMAT` describing the column's data type, and then a `uintN_t` to say how many columns the accessor has